crates/typst-syntax/src/span.rs

```diff
@@ -41,7 +45,7 @@ impl Span {
     ///
     /// Returns `None` if `number` is not contained in `FULL`.
     pub(super) const fn new(id: FileId, number: u64) -> Option<Self> {
-        if number < Self::FULL.start || number >= Self::FULL.end {
+        if !contains(Self::FULL, number) {
```

Comment:
IMHO unnecessary unless `contains` would be used more than once. A short comment above the `if` would suffice.

Reply:
Um, `Range` already has a `contains` method. `Self.FULL.contains(&number)`